File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
import sys

# splitlines()로 빈 줄 자동으로 처리
input = sys.stdin.read().splitlines()

n = int(input[0])
data = []

for i in range(1, n + 1):
    m = input[i].split()
    age = int(m[0])
    name = m[1]
    data.append([age, name])

data.sort(key=lambda x: x[0])

for age, name in data:
    print(age, name)

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : 'input.txt').toString().trim().split('\n');

const N = Number(input[0]);

const users = input.slice(1).map((line, idx) => {
    const [age, name] = line.split(' ');
    return { age: Number(age), name, index: idx };
});

users.sort((a, b) => {
    if (a.age !== b.age) return a.age - b.age;
    return a.index - b.index; // 입력 순서 유지하기 위해
});

let answer = '';
for (let user of users) {
    answer += `${user.age} ${user.name}\n`;
}

console.log(answer.trim());